packages/compile-solidity/src/run.ts

```diff
@@ -124,8 +144,8 @@ function orderABI({ abi, contractName, ast }) {
 
   // Put function names in a hash with their order, lowest first, for speed.
   const functionIndexes = orderedFunctionNames
-    .map((functionName, index) => ({ [functionName]: index }))
-    .reduce((a, b) => Object.assign({}, a, b), {});
+    .map((functionName: string, index: number) => ({ [functionName]: index }))
+    .reduce((a: object, b: object) => Object.assign({}, a, b), {});
```

Comment:
I don't think `object` is an appropriate type here, is it? That is, shouldn't it be an object with the same index signature that was defined by the map?

```suggestion
    .reduce((a: { [functionName: string]: number } , b: { [functionName: string]: number }) => Object.assign({}, a, b), {});
```

Reply:
Ok, that is a bit more precise.